Adds ability to update pull request branch

bjedrocha opened the pull request:
Resolves #1471 

As per https://docs.github.com/en/rest/pulls/pulls#update-a-pull-request-branch. I couldn't get the specs to work with VCR despite creating a test repo and setting all relevant ENV vars. I decided to stub the request instead.

timrogers commented on lib/octokit/client/pull_requests.rb:
```diff
@@ -276,6 +276,17 @@ def pull_request_files(repo, number, options = {})
+      # @param options [Hash] Optional parameters (e.g. expected_head_sha)
+      # @return [Sawyer::Resource] Info regarding the update if successful
+      def update_pull_request_branch(repo, number, options = {})
+        put "#{Repository.path repo}/pulls/#{number}/update-branch", options
```
What do you thinking of using `#boolean_from_response` here? Looking at the API, it appears that the response is not super useful - you just want to know if it worked or not.

```suggestion
        boolean_from_response(:put, "#{Repository.path repo}/pulls/#{number}/update-branch", options)
```

timrogers commented on spec/octokit/client/pull_requests_spec.rb:
```diff
@@ -136,6 +136,15 @@
   end # new @pull methods
 
+  # stub this so we don't have to set up new fixture data
+  describe '.update_pull_request_branch' do
```
We should test the return value of `#update_pull_request_branch` in here.

timrogers reviewed (changes requested):
👋🏻 Thanks for taking the time to make this PR! I have a suggestion on how we could make the behaviour of the method a little nicer, and on how we test it.

bjedrocha commented:
Thanks for the quick review - I can address that feedback shortly. What are your thoughts on having the API called stubbed vs. having a cassette for it?

timrogers commented:
> Thanks for the quick review - I can address that feedback shortly. What are your thoughts on having the API called stubbed vs. having a cassette for it?

I'm okay for it to be stubbed. Could we stub it with an actual response for better test. overage?

bjedrocha commented:
Addressed your feedback. I've changed `update_pull_request_branch` to return a simple boolean if the request succeeds. I also improved the test coverage by adding specs for the requests that fail. I didn't stub the response body as we're not doing anything with anyway but that can always be added if you think it provides value.

tride1997 commented:
1472

Vào 23:32, Th 3, 23 thg 8, 2022 Bart Jedrocha ***@***.***> đã
viết:

> Addressed your feedback. I've changed update_pull_request_branch to
> return a simple boolean if the request succeeds. I also improved the test
> coverage by adding specs for the requests that fail. I didn't stub the
> response body as we're not doing anything with anyway but that can always
> be added if you think it provides value.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/octokit/octokit.rb/pull/1472#issuecomment-1224315252>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMLANDXD7NGCF5C6I6VJ62TV2T4I3ANCNFSM57IIZBNQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
